cpp/examples/minimal_build/CMakeLists.txt

```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.25)
```

Comment:
Can we keep using 3.16?
We don't need to require CMake 3.25 or later for users who just use `find_package(Arrow)`.

Reply:
sounds reasonable to me. I've reverted.